### ISSUE TITLE
'smart is not a valid ColonReplacementFormat' when creating radarr_naming resource

### ISSUE THREAD

**Issue by tinhtruong** [devopsarr/terraform-provider-radarr#273]
When I've tried to create resources as following:

```terraform
resource "radarr_naming" "default" {
  rename_movies              = true
  replace_illegal_characters = false
  colon_replacement_format   = "spaceDashSpace"
  standard_movie_format      = "{Movie Title} ({Release Year}) {Quality Full}"
  movie_folder_format        = "{Movie Title} ({Release Year})"
}
```

I've got the error:

```shell
FAILED! => {"changed": false, "cmd": "/usr/bin/terraform apply -no-color -input=false -auto-approve -lock=true /tmp/tmpb93jw1t1.tfplan", "msg": "\nError: Client Error\n\n  with radarr_naming.default,\n  on main.tf line 40, in resource \"radarr_naming\" \"default\":\n  40: resource \"radarr_naming\" \"default\" {\n\nUnable to read naming, got error: smart is not a valid ColonReplacementFormat\nDetails:\n{\n  \"renameMovies\": false,\n  \"replaceIllegalCharacters\": true,\n  \"colonReplacementFormat\": \"smart\",\n  \"standardMovieFormat\": \"{Movie Title} ({Release Year}) {Quality Full}\",\n  \"movieFolderFormat\": \"{Movie Title} ({Release Year})\",\n  \"id\": 1\n}", "rc": 1, "stderr": "\nError: Client Error\n\n  with radarr_naming.default,\n  on main.tf line 40, in resource \"radarr_naming\" \"default\":\n  40: resource \"radarr_naming\" \"default\" {\n\nUnable to read naming, got error: smart is not a valid ColonReplacementFormat\nDetails:\n{\n  \"renameMovies\": false,\n  \"replaceIllegalCharacters\": true,\n  \"colonReplacementFormat\": \"smart\",\n  \"standardMovieFormat\": \"{Movie Title} ({Release Year}) {Quality Full}\",\n  \"movieFolderFormat\": \"{Movie Title} ({Release Year})\",\n  \"id\": 1\n}\n", "stderr_lines": ["", "Error: Client Error", "", "  with radarr_naming.default,", "  on main.tf line 40, in resource \"radarr_naming\" \"default\":", "  40: resource \"radarr_naming\" \"default\" {", "", "Unable to read naming, got error: smart is not a valid ColonReplacementFormat", "Details:", "{", "  \"renameMovies\": false,", "  \"replaceIllegalCharacters\": true,", "  \"colonReplacementFormat\": \"smart\",", "  \"standardMovieFormat\": \"{Movie Title} ({Release Year}) {Quality Full}\",", "  \"movieFolderFormat\": \"{Movie Title} ({Release Year})\",", "  \"id\": 1", "}"], "stdout": "radarr_naming.default: Creating...\nradarr_download_client_sabnzbd.sabnzbd[0]: Creating...\nradarr_media_management.default: Creating...\nradarr_root_folder.movies: Creating...\nradarr_media_management.default: Creation complete after 1s\nradarr_root_folder.movies: Creation complete after 1s\nradarr_download_client_sabnzbd.sabnzbd[0]: Creation complete after 1s [name=sabnzbd]\n", "stdout_lines": ["radarr_naming.default: Creating...", "radarr_download_client_sabnzbd.sabnzbd[0]: Creating...", "radarr_media_management.default: Creating...", "radarr_root_folder.movies: Creating...", "radarr_media_management.default: Creation complete after 1s", "radarr_root_folder.movies: Creation complete after 1s", "radarr_download_client_sabnzbd.sabnzbd[0]: Creation complete after 1s [name=sabnzbd]"]}
```

My environment:
Terraform version: v1.10.0
Radarr version: 5.15.1.9463-2
devopsarr/radarr: 2.3.0

Any help would be appreciated

**Comment by tinhtruong** [comment]
I just took a look at the latest Radarr API spec and it changed to include a new possible value 'smart'
https://radarr.video/docs/api/#/NamingConfig/get_api_v3_config_naming_examples

**Comment by Fuochi** [comment]
Hi @tinhtruong, thanks for opening the bug, you are right, the value has been recently introduced, and the same is happening already in #269's tests. The SDK radarr-go already implements the new feature. However, I'll have to fix and merge #270 to address the issue.
Unfortunately, this time of the year is very busy, I'll try to do my best to address this as soon as I can.

Meanwhile, since the error is happening during the read, if you manage to manually change the first time `ColonReplacementFormat` to an old valid value like `spaceDashSpace`, either via GUI or API. I suppose it would then work for the successive TF runs.

**Comment by Fuochi** [comment]
Hi @tinhtruong, please test v2.3.1 and let me know if it works
